[FIX] runbot_gitlab: Retrieve correctly SSH keys when under GitLab CI

luisg123v opened the pull request:
Currently, SSH keys are not retrieved correctly, due to a bad parsing of
the HTTP response. When keys are retrieved from GitLab, the response's
content is not a json, but a raw page; so it's raw bites, not str.

This commit parses correctly such responses, so SSH keys may be parsed
correctly and SSH access may be granted.

This also causes empty authors to be skipped, they of course have
neither username nor SSH keys, and retrieving them from GitLab would
cause issues.

Closes https://github.com/Vauxoo/runbot-addons/issues/173

luisg123v commented:
@moylop260  
Could you review and merge this, please?

I'm pretty confident this is the actual fix, or at least part of it.

moylop260 commented on runbot_travis2docker/models/runbot_build.py:
```diff
@@ -185,14 +185,15 @@ def _get_ssh_keys(self):
         keys = ""
         for own_key in ['author', 'committer']:
+            login = response.get(own_key, {}).get('login')
+            if login == '':
```
if not login
In order to apply any Faly value like None, False, “”, 0

luisg123v replied:
I decided to do it that way to ensure only empty authors are skipped. Any other falsy value, e.g. `None` would mean any error occurred, which shouldn't be ignored IMO.

moylop260 commented on runbot_travis2docker/models/runbot_build.py:
```diff
@@ -185,14 +185,15 @@ def _get_ssh_keys(self):
             return
         keys = ""
         for own_key in ['author', 'committer']:
+            login = response.get(own_key, {}).get('login')
```
If the response[own_key] is None you will have a NoneType error.
You can use
(response.get(own_key) or {}).get(...

luisg123v replied:
Yes, you're right, altough it would be `AttributeError`.

There is another possible issue, the key might not be a dictionary, which would also raise `AttributeError`, so I think I will enclose that sentence in the below try-catch block.

luisg123v replied:
@moylop260  
I did: `login = response.get(own_key).get('login')`
and added `AttributeError` to the try-except. What do you think?

I'd like to add, I think none of those possibilities should be possible, unless GitLab API is broken. But, since it's a json, `None` and an empty list could be theoretically possible values.

luisg123v committed:
[FIX] runbot_gitlab: Retrieve correctly SSH keys when under GitLab CI

Currently, SSH keys are not retrieved correctly, due to a bad parsing of
the HTTP response. When keys are retrieved from GitLab, the response's
content is not a json, but a raw page; so it's raw bites, not str.

This commit parses correctly such responses, so SSH keys may be parsed
correctly and SSH access may be granted.

This also causes empty authors to be skipped, they of course have
neither username nor SSH keys, and retrieving them from GitLab would
cause issues.
